### PR TITLE
Allow some candidates to re-subscribe for a TTA

### PR DESCRIPTION
**Pull request by ethax-ross**
> :warning: **Do not merge until CRM team have tested on the review instance**

### Trello card

[Trello-1698](https://trello.com/c/yG3hzAuh/1698-development-unblock-get-an-adviser-applicants-who-have-previously-had-an-adviser)

### Context

Previously we were using the `already_subscribed_to_teacher_training_adviser` field to determine if we prevent the user from continuing with their sign up. Instead, we want to use the `can_subscribe_to_teacher_training_adviser` field which will allow some candidates that have already subscribed to re-subscribe (depending on their adviser status as determined by the
API).

### Changes proposed in this pull request

- Bump API Client version
- Allow some candidates to re-sign up for a TTA

### CODE REVIEW

**Review by MylesJarvis — COMMENTED** [review]
Going to remove myself on this and allow for a dev review

**Comment by MylesJarvis** [comment]
Never mind I can't as code owner - will await a dev review and then approve after

**Review by leoapost — APPROVED** [review]
Looks ok to me

**Review by leoapost — APPROVED** [review]

**Review by leoapost — DISMISSED** [review]

**Review by leoapost — APPROVED** [review]

**Pull request merged**
